### ISSUE TITLE
iOS 14.2 (physical device) - getSSID always nil

### ISSUE THREAD

**Issue by DominikStarke** [alternadom/WiFiFlutter#105]
Hi,
I'm trying to join a wifi network using this plugin, which works.
However connect always returns false, despite successfully connecting. I have narrowed the issue down to the swift implementation of getSSID to always return nil. More specifically CNCopySupportedInterfaces returning nil.

My entitlements contain these:
```
	<key>com.apple.developer.networking.HotspotConfiguration</key>
	<true/>
	<key>com.apple.developer.networking.networkextension</key>
	<array>
		<string>app-proxy-provider</string>
	</array>
	<key>com.apple.developer.networking.wifi-info</key>
	<true/>
	<key>com.apple.external-accessory.wireless-configuration</key>
	<true/>
```

My plist also contains access to location:
```
	<key>NSLocationAlwaysUsageDescription</key>
	<string>???</string>
	<key>NSLocationWhenInUseUsageDescription</key>
	<string>???</string>
	<key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
	<string>???</string>
	<key>NSLocationUsageDescription</key>
	<string>???</string>
	<key>NSLocalNetworkUsageDescription</key>
	<string>???</string>
```
When I open iOS Settings location usage is set to grant always.
I'm at my whits end. Am I missing something?
Thanks in advance!

**Comment by DominikStarke** [comment]
Resolved after rebooting the phone...
I'll leave this open for now, as I'm not entirely sure my setup is correct.

**Comment by daadu** [comment]
Did you face this again? Check https://stackoverflow.com/questions/56583650/cncopycurrentnetworkinfo-with-ios-13 for the explaination. Any idea of add fix to the current implementation. `getSSID` and `isEnabled` both use it.

**Comment by DominikStarke** [comment]
Not really, it appears that changing the phone's privacy settings only take full effect after rebooting (under some circumstances?). Otherwise it might somehow mess with other privacy related stuff like CNCopyX.

But that's a hard guess.

**Comment by daadu** [comment]
New API to get  current network info (linked after iOS 12), mentioned here - https://developer.apple.com/reference/systemconfiguration/1614126-cncopycurrentnetworkinfo

**Comment by daadu** [comment]
The fix has landed on pub.dev with v0.3.4.

@DominikStarke Thanks for reporting this bug. Check yourself in README!!

**Comment by DominikStarke** [comment]
This issues seems to have resurfaced with the use of `NEHotspotNetwork.fetchCurrent`
With the entitlements and usage descriptions from the initial post I can't get getSSID to work on iOS 14.4.2 (physical device).

The location permissions are set to always and allow fine location.

CNCopySupportedInterface still works, so if I comment out
```
    if #available(iOS 14.0, *) {
        NEHotspotNetwork.fetchCurrent(completionHandler: { currentNetwork in
            ssid = currentNetwork?.ssid
        })
    } 
```

I also tried SwiftWifiIotPlugin to directly implement CLLocationManagerDelegate which properly requested the permission, but it still didn't work.

**Comment by daadu** [comment]
This has landed on pub.dev with [v0.3.7](https://pub.dev/packages/wifi_iot/versions/0.3.7)
